### ISSUE TITLE
Getting build error while building git-crypt on a docker based on ubuntu:14.04

### ISSUE THREAD

**Issue by richtera** [AGWA/git-crypt#111]
```g++
g++ -Wall -pedantic -Wno-long-long -O2 -Wno-deprecated-declarations   -c -o crypto-openssl.o crypto-openssl.cpp
crypto-openssl.cpp:74:11: error: field ‘ctx’ has incomplete type ‘HMAC_CTX {aka hmac_ctx_st}’
  HMAC_CTX ctx;
           ^~~
In file included from /usr/include/openssl/evp.h:14:0,
                 from /usr/include/openssl/hmac.h:15,
                 from crypto-openssl.cpp:36:
/usr/include/openssl/ossl_typ.h:102:16: note: forward declaration of ‘HMAC_CTX {aka struct hmac_ctx_st}’
 typedef struct hmac_ctx_st HMAC_CTX;
                ^~~~~~~~~~~
crypto-openssl.cpp: In destructor ‘Hmac_sha1_state::~Hmac_sha1_state()’:
crypto-openssl.cpp:88:27: error: ‘HMAC_cleanup’ was not declared in this scope
  HMAC_cleanup(&(impl->ctx));
                           ^
<builtin>: recipe for target 'crypto-openssl.o' failed
```
Unfortunately I am not quite sure how to track down a fix.

**Comment by AGWA** [comment]
What version of OpenSSL are you building against?  Note that git-crypt doesn't support 1.1 yet.

**Comment by richtera** [comment]
Got it. That's got to be it

**Comment by kalbasit** [comment]
@AGWA are you planning on porting git-crypt to openssl-1.1?

**Comment by artem-sidorenko** [comment]
@richtera I'm building the git-crypt packages which are missing in distros via [OBS](https://software.opensuse.org//download.html?project=home%3Aartem_sidorenko&package=git-crypt), maybe this helps

**Comment by richtera** [comment]
I just switched distro and was fine. But I think it does need to be addressed eventually. I have not played with OBS recently.

**Comment by buchuki** [comment]
This is breaking git-crypt under the new Fedora 26 beta.

**Comment by florenthemmi** [comment]
Not compiling since Debian 9.0

```
> openssl version
OpenSSL 1.1.0f  25 May 2017
```

**Comment by AGWA** [comment]
Closed in 553c1b03876e174b117bb5250593217418f425a4
